### PR TITLE
actions: pre-commit and opentelemetry exporter

### PR DESCRIPTION
**Pull request by v1v**
## What does this PR do?

Use GitHub actions for the `pre-commit` and export data to OpenTelemetry
Remove `pre-commit` stage in the CI Jenkinsfile and remove meta-stage, so the `Unit Test` stage runs normally

## Why is it important?

Simplify the CI pipeline

### CODE REVIEW

File: .github/workflows/pre-commit.yml [hunk 1]
```diff
@@ -0,0 +1,35 @@
+---
+name: pre-commit
+
+on:
+  pull_request:
```
Comment (v1v) [inline .github/workflows/pre-commit.yml:5]:
```suggestion
  push:
    branches: [main]
  pull_request:
```

**Review by mdelapenya — APPROVED** [review]
Super cool!

Have you thought about running the unit tests in the GH action too?

**Review by kuisathaverat — APPROVED** [review]

File: .github/workflows/pre-commit.yml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+    # Uses Go version from the repository.
+    - name: Read .go-version file
+      id: goversion
+      run: echo "::set-output name=version::$(cat .go-version)"
+    - uses: actions/setup-go@v3
+      with:
+        go-version: "${{ steps.goversion.outputs.version }}"
```
Comment (v1v) [inline .github/workflows/pre-commit.yml:24]:
```suggestion
    - uses: actions/setup-go@v3
      with:
        go-version-file: .go-version
```

File: .github/workflows/pre-commit.yml [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+        extra_args: --help
+    - id: files
+      uses: jitterbit/get-changed-files@b17fbb00bdc0c0f63fcf166580804b4d2cdc2a42
+    - name: Install dependencies
+      run: .ci/scripts/install-dependencies.sh
+    - name: Precommit changes
+      run: |
+        for changed_file in ${{ steps.files.outputs.all }}; do
+          pre-commit run --files "${changed_file}"
+        done
```
Comment (v1v) [inline .github/workflows/pre-commit.yml:37]:
```suggestion
    - name: Install dependencies
      run: .ci/scripts/install-dependencies.sh
    - uses: elastic/apm-pipeline-library/.github/actions/pre-commit@current
```

**Pull request merged**